snapshot/restore/shallow_fs_output.go

```diff
@@ -64,6 +64,12 @@ func (o *ShallowFilesystemOutput) WriteFile(ctx context.Context, relativePath st
 		return o.FilesystemOutput.WriteFile(ctx, relativePath, f)
 	}
 
+	// Always restore .kopiaignore files so that they correctly apply to
+	// later snapshots.
+	if f.Name() == ".kopiaignore" {
```

Comment:
Unfortunately `.kopiaignore` is not hardcoded, by settable by policies and there could be several filenames here.

You probably want to restore all files listed in `FilesPolicy.DotIgnoreFiles` effective for the directory we're restoring to (so when the directory is later snapshotted, we will consider the correct ones). This is actually quite complicated in terms of layering (output probably should not depend on source policies), so we may need to introduce some helper indirection

Reply:
Thanks for the insight. I will adjust.